build/unix/makeversion.py

```diff
@@ -0,0 +1,64 @@
+# Script to update base/inc/RVersion.h.
+# Called by main Makefile as soon as build/version_number has been updated.
+#
+# Author: Axel, 2020-03-06
+
+import os, subprocess, re
+from datetime import date, datetime
+
+versionline = ""
+with open("build/version_number", "r") as file:
+  versionline = file.read().replace('\n', '')
+
+matches = re.match(r'^(\d+)[.](\d+)/(\d+)$', versionline).groups()
+if len(matches) != 3:
+  raise RuntimeError("build/version_number: invalid syntax")
+
+major = int(matches[0])
+minor = int(matches[1])
+patch = int(matches[2])
+vers_code = (major << 16) + (minor << 8) + patch
+
+datenow = date.today().strftime("%b %d %Y") # Sep 11 2019
+timenow = datetime.now().strftime("%H:%M:%S") # 15:05:55
+
+sourcecode = """#ifndef ROOT_RVersion
+#define ROOT_RVersion
+
+/* Version information automatically generated by installer. */
+
+/*
+ * These macros can be used in the following way:
+ *
+ *    #if ROOT_VERSION_CODE >= ROOT_VERSION(6,32,4)
+ *       #include <newheader.h>
+ *    #else
+ *       #include <oldheader.h>
+ *    #endif
+ *
+*/
+
+#define ROOT_RELEASE "{}"
+#define ROOT_RELEASE_DATE "{}"
+#define ROOT_RELEASE_TIME "{}"
+#define ROOT_VERSION(a,b,c) (((a) << 16) + ((b) << 8) + (c))
+#define ROOT_VERSION_CODE ROOT_VERSION({},{},{}) /* {} */
+
+#endif
+""".format(versionline, datenow, timenow, major, minor, patch, vers_code)
+
+with open('core/base/inc/RVersion.h', 'w') as file:
+  file.write(sourcecode)
+
+subprocess.check_call("build/unix/coreteam.sh rootx/src/rootcoreteam.h", shell = True)
```

Comment:
This won't work on Windows anyway, so I don't see why Python is better than bash for this.

Reply:
why it will not work?

Reply:
..actually it was never running before on windows: https://github.com/root-project/root/pull/5095/files#diff-af3b638bc2a3e6c650974192a53c7291R452

Reply:
> I don't see why Python is better than bash for this.

Error handling is my only reason. And I now have a python script, so writing it in bash would be strictly more work :-)

Reply:
> why it will not work?

Ah, I see shell script... Anyway it was never working on windows. 

It is up to @axel to decide the fate of PR since he is a release manager. (for me it looks good)

Reply:
This looks like you could simply have a "RVersion.h.in" and use CMake variable substitution and avoid both bash and python. But, as you said, you already wrote it in Python, might as well just use that for now.

Reply:
CMake: yep, I realized that once you said "why not bash?" Next time!